specs/data_structures.md

```diff
@@ -224,49 +225,91 @@ Merkle trees are used to authenticate various pieces of data across the LazyLedg
 
 ## Binary Merkle Tree
 
-Binary Merkle trees are constructed in the usual fashion, with leaves being hashed once to get leaf nodes and internal nodes being the hash of the concatenation of their children. Note that when hashing leaves , `0x00` is prepended, and when hashing internal nodes, `0x01` is prepended. This avoids a second-preimage attack [where internal nodes are presented as leaves](https://en.wikipedia.org/wiki/Merkle_tree#Second_preimage_attack).
+Binary Merkle trees are constructed in the usual fashion, with leaves being hashed once to get leaf node values and internal node values being the hash of the concatenation of their children.
```

Comment:
While there might be an argument against differentiating internal and leaf nodes with a prefix in the sparse Merkle tree case (although all major SMT implementations still do that), this section speak about general binary trees. If you do not want to explicitly mention porefixing here, maybe say that leaf nodes are hashed with a hash function hash_leaf and inner nodes are hashed with another one hash_inner. In the case of no prefixes both are the same.